src/python/pants/backend/codegen/protobuf/target_types.py

```diff
@@ -27,3 +39,24 @@ class ProtobufLibrary(Target):
     alias = "protobuf_library"
     core_fields = (*COMMON_TARGET_FIELDS, ProtobufDependencies, ProtobufSources, ProtobufGrpcToggle)
     help = f"Protobuf files used to generate various languages.\n\nSee f{doc_url('protobuf')}."
+
+
+class GenerateProtobufLibraryFromProtobufLibrary(GenerateTargetsRequest):
+    generate_from = ProtobufLibrary
```

Comment:
Nit: maybe `GenerateProtobufLibraries` or `GenerateFileTargets`? I get what you're going for here, but even if/when we generate a different target type for children, the name would be unwieldy.

It also seemingly leaves room for multiple target generators per target type, and that probably shouldn't be legal.